abnf/odata-abnf-construction-rules.txt

```diff
@@ -389,15 +395,21 @@ selectItem     = STAR
                      / optionallyQualifiedFunctionName
                      )
 selectProperty = primitiveProperty / primitiveAnnotationInQuery
-               / ( primitiveColProperty / primitiveColAnnotationInQuery ) [ OPEN selectOptionPC *( SEMI selectOptionPC ) CLOSE ]
+               / ( primitiveColProperty / primitiveColAnnotationInQuery ) 
+                 [ count [ OPEN countOption *( SEMI countOption ) CLOSE ] 
+                   / [ OPEN colOption *( SEMI colOption ) CLOSE ] 
+                 ]
                / navigationProperty
-               / selectPath [ OPEN selectOption *( SEMI selectOption ) CLOSE
-                            / "/" selectProperty
+               / selectPath [ OPEN structOption *( SEMI structOption ) CLOSE
+                              / "/" selectProperty
+                            ]
+               / selectColPath [ count [ OPEN countOption *( SEMI countOption ) CLOSE ]
+                              / OPEN selectOption *( SEMI selectOption ) CLOSE
+                              / "/" selectProperty
                             ]
-selectPath     = ( complexProperty / complexColProperty / complexAnnotationInQuery ) [ "/" optionallyQualifiedComplexTypeName ]
-selectOptionPC = filter / search / inlinecount / orderby / skip / top
-selectOption   = selectOptionPC
-               / compute / select / aliasAndValue
+selectPath     = ( complexProperty / complexAnnotationInQuery ) [ "/" optionallyQualifiedComplexTypeName ]

```

Comment:
What if `complexAnnotationInQuery` is collection-valued (analogous to `entityAnnotationInQuery`)? Shouldn't this be part of `selectColPath`?

Or shall we introduce all four?
- `entityAnnotationInQuery`
- `entityColAnnotationInQuery`
- `complexAnnotationInQuery`
- `complexColAnnotationInQuery`

Reply:
should handle consistently with expand.  ideally, define both complexAnnotationInQuery and complexColAnnotationInQuery.

Reply:
added, and used, entityColAnnotationInQuery and complexColAnnotationInQuery.

Reply:
Fixed.